src/Integration.Vsix/CFamily/PreCompiledHeaders/PreCompiledHeadersEventListener.cs

```diff
@@ -71,12 +71,17 @@ internal sealed class PreCompiledHeadersEventListener : IPreCompiledHeadersEvent
 
             pchJobTimeoutInMilliseconds = environmentSettings.PCHGenerationTimeoutInMs(60 * 1000);
 
-            activeDocumentTracker.OnDocumentFocused += OnActiveDocumentFocused;
+            activeDocumentTracker.ActiveDocumentChanged += OnActiveDocumentFocused;
         }
 
-        private void OnActiveDocumentFocused(object sender, DocumentFocusedEventArgs e)
+        private void OnActiveDocumentFocused(object sender, ActiveDocumentChangedEventArgs e)
         {
-            var detectedLanguages = sonarLanguageRecognizer.Detect(e.TextDocument.FilePath, e.TextDocument.TextBuffer.ContentType);
+            if (e.ActiveTextDocument == null)
+            {
+                return;
+            }
```

Comment:
This is the only substantive code change. Previously, this class relied on the active document not being null, so now it has to handle that case gracefully.